pkg/connection/manager.go

```diff
@@ -133,28 +135,110 @@ func (m *DetailsManager) publishConnection(ctx context.Context, so SecretOwner,
 		return errors.Wrap(err, errConnectStore)
 	}
 
-	return errors.Wrap(ss.WriteKeyValues(ctx, store.Secret{
-		Name:     p.Name,
-		Scope:    so.GetNamespace(),
+	err = ss.DeleteKeyValues(ctx, &store.Secret{
+		ScopedName: store.ScopedName{
+			Name:  p.Name,
+			Scope: so.GetNamespace(),
+		},
 		Metadata: p.Metadata,
-	}, kv), errWriteStore)
+		Data:     store.KeyValues(conn),
+	})
```

Comment:
Would using `SecretMustBeOwnedBy` make sense here too, to make sure we don't delete a secret we don't own?

Reply:
Yes, good idea, done!